src/frontend/views/games_view.py

```diff
@@ -74,6 +59,34 @@ def get_context_data(self, **kwargs):
         context["sgfplayer_url_name"] = "sgfplayer_rating" # urls.py
       return context
 
+class GamesListByUserView(ListView):
+  template_name = "pages/games_list_by_user.html"
+  paginate_by = 100
+
+  def get_queryset(self):
+    self.run = get_object_or_404(Run, name=self.kwargs["run"])
+    self.user = get_object_or_404(User, username=self.kwargs["user"])
+
+    if self.kwargs["kind"] == "training":
+      games = TrainingGame.objects.filter(submitted_by=self.user,run=self.run).order_by("-created_at")
+    else:
+      games = RatingGame.objects.filter(submitted_by=self.user,run=self.run).order_by("-created_at")
+    return games
+
+  def get_context_data(self, **kwargs):
+      context = super().get_context_data(**kwargs)
+      context["run"] = self.run
+      context["user"] = self.user
+      context["kind"] = self.kwargs["kind"]
+      if self.kwargs["kind"] == "training":
+        context["list_url_name"] = "training_games_list_by_user" # urls.py
+        context["sgfplayer_url_name"] = "sgfplayer_training" # urls.py
+      else:
+        context["list_url_name"] = "rating_games_list_by_user" # urls.py
+        context["sgfplayer_url_name"] = "sgfplayer_rating" # urls.py
+      return context
```

Comment:
Same than above, we are having a lot of if, in view, template. We should have two views, two templates and no if, then extract the similar parts.